src/interfaces/modular/shogun-base.i.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# Written (W) 2016 Sanuj Sharma
+
+def entry(templateFile, class_list_file):
+    templateLoader = jinja2.FileSystemLoader(searchpath="/")
+    templateEnv = jinja2.Environment(loader=templateLoader)
+
+    template = templateEnv.get_template(templateFile)
+
+    classes = []
+    with open(class_list_file) as f:
+        for line in f:
+            if line[0] is 'C' or line[0] is 'T':
+                temp = [elt.strip() for elt in line.split(',')]
+                classes.append(temp)
+
+
+    types = ['PT_BOOL', 'PT_CHAR', 'PT_INT8', 'PT_UINT8', 'PT_INT16', 'PT_UINT16', 'PT_INT32',
+             'PT_UINT32', 'PT_INT64', 'PT_UINT64', 'PT_FLOAT32', 'PT_FLOAT64', 'PT_FLOATMAX']
+
+    templateVars = {"classes" : classes, "types" : types}
+
+    return template.render(templateVars)
+
+# execution
+# ./shogun-base.i.py <template file> <output file name> <extra args...>
+
+import sys
+TEMPLATE_FILE = sys.argv[1]
+output_file = sys.argv[2]
+class_list_file = sys.argv[3]
+
+try:
+    import jinja2
+    outputText = entry(TEMPLATE_FILE, class_list_file)
+except ImportError:
+    import os
+    basename = os.path.basename(output_file)
+    basename = basename.replace('.cc', '')
+    print "Please install jinja2 for shogun-base.i";
```

Comment:
Can we please integrate this into cmake properly rather than printing python output and then not failing?